server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/DelaysAndMaxRetry.java

```diff
@@ -64,12 +64,13 @@ private static DelaysAndMaxRetry getDelaysAndMaxRetry(int intendedMaxRetries, in
     private static DelaysAndMaxRetry addExtraAttemptToLastDelay(int intendedMaxRetries, int extra, List<Delay> delayTimesList) throws MessagingException {
         if (delayTimesList.size() != 0) {
             Delay lastDelay = delayTimesList.get(delayTimesList.size() - 1);
-            LOGGER.warn("Delay of {} msecs is now attempted: {} times", lastDelay.getDelayTime(), lastDelay.getAttempts());
+            Duration lastDelayTime = lastDelay.getDelayTime();
+            LOGGER.warn("Delay of {} msecs is now attempted: {} times", lastDelayTime.toMillis(), lastDelay.getAttempts());
```

Comment:
I wouldn't have express the delay in msecs. Duration may have a human-readable formater?

Reply:
The Duration.toString() returns a string kind of “P80S...” something.... It overrides toString() method, but impossible to understand. I will use some approprirate duration formaters

Reply:
Can we use http://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/time/DurationFormatUtils.html ?